independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/MemoryClassPathElement.java

```diff
@@ -22,7 +23,21 @@ public MemoryClassPathElement(Map<String, byte[]> resources) {
     }
 
     public void reset(Map<String, byte[]> resources) {
-        this.resources = resources;
+        Map<String, byte[]> newResources = new HashMap<>(resources);
+        //we can't delete .class files from the loader
+        //gizmo may not generate the same function names on restart
+        //so if we delete them already loaded classes may have problems, as functions they reference
```

Comment:
So if I understand it correctly when we generate a fwk class that is not marked as an application class it's loaded by the "Base Runtime ClassLoader" which is persistent but always reset during restart (because the generated classes may differ). But even if gizmo generates a different name for a function the new generated fwk class should reference this new name, or? I mean the generated class cannot not survive the hot reload, can it?

Reply:
gizmo is not generating stable function names, so if you generate a new class that is exactly the same the function names will be different. If the outer class has been loaded but not the function then the original function would get deleted by the reset code.

Reply:
Really gizmo should generate stable function names, but this is safer regardless.

Reply:
> If the outer class has been loaded but not the function...

I see. BTW what happens if the outer class is loaded but should be discarded because the newly generated class differs? I'm asking because the base runtime CL is said to be persistent...

Reply:
Then we are doing something wrong. Base runtime classes should be effectivly immutable, e.g. a client proxy for a bean in the base runtime CL will never change, as the class it is based on will never change. It might be added though as the bean may have been removed, but the user might add something to change that. The reset mechanism is really supposed to be about adding new classes.

Reply:
Now I finally understand ;-). Thanks! In our case (#8301), the generated value resolver should be identical.